src/sklearn_compat/_sklearn_compat.py

```diff
@@ -137,14 +118,93 @@ def _to_new_tags(old_tags, estimator=None):
         classifier_tags=classifier_tags,
         regressor_tags=regressor_tags,
         input_tags=input_tags,
-        array_api_support=old_tags["array_api_support"],
+        # Array-API was introduced in 1.3, we need to default to False if not inside
+        # the old-tags.
+        array_api_support=old_tags.get("array_api_support", False),
         no_validation=old_tags["no_validation"],
         non_deterministic=old_tags["non_deterministic"],
         requires_fit=old_tags["requires_fit"],
         _skip_test=old_tags["_skip_test"],
     )
 
 
+########################################################################################
+# Upgrading for scikit-learn 1.3
+########################################################################################
+
+if sklearn_version < parse_version("1.3"):
+    # parameter validation
+    class ParamsValidationMixin:
+        """Mixin class to validate parameters.
+
+        For scikit-learn < 1.3, we do not provide any validation since the parameter
+        validation is not implemented as part of scikit-learn. We would need to backport
+        the infrastructure. Since this infrastructure required to look at the global
+        configuration of scikit-learn, we prefer to not interfere and just pass-by.
+        """
+
+        def _validate_params(self):
+            """No-op."""
+            pass
+
+    def _fit_context(*, prefer_skip_nested_validation):
+        """Decorator to run the fit methods of estimators within context managers.
+
+        With scikit-learn < 1.3, this decorator is no-op.
+
+        Parameters
+        ----------
+        prefer_skip_nested_validation : bool
+            If True, the validation of parameters of inner estimators or functions
+            called during fit will be skipped.
+
+            This is useful to avoid validating many times the parameters passed by the
+            user from the public facing API. It's also useful to avoid validating
+            parameters that we pass internally to inner functions that are guaranteed to
+            be valid by the test suite.
+
+            It should be set to True for most estimators, except for those that receive
+            non-validated objects as parameters, such as meta-estimators that are given
+            estimator objects.
+
+        Returns
+        -------
+        decorated_fit : method
+            The decorated fit method.
+        """
+
+        def decorator(fit_method):
+            @functools.wraps(fit_method)
+            def wrapper(estimator, *args, **kwargs):
+                estimator._validate_params()
+                return fit_method(estimator, *args, **kwargs)
+
+            return wrapper
+
+        return decorator
+else:
+    # parameter validation
+    class ParamsValidationMixin:
+        """Mixin class to validate parameters."""
```

Comment:
add a comment that it's here cause in sklearn we have in in `BaseEstimator`?

We also need to make sure (at least in our README) that the inheritance is of `class Estimator(..., BaseEstimator, ParamsValidationMixin):` so that in new sklearn versions `ParamsValidationMixin` is mute.